gordo/machine/machine.py

```diff
@@ -118,3 +118,13 @@ def to_dict(self):
             "project_name": self.project_name,
             "evaluation": self.evaluation,
         }
+
+    def report(self):
+        """ 
+        Run any reporters in the machine's runtime for the current state.
+        """
+        # Avoid circular dependency with reporters which import Machine
+        from gordo.reporters.base import BaseReporter
+
+        for reporter in map(BaseReporter.from_dict, self.runtime["reporters"]):
+            reporter.report(self)
```

Comment:
What about the "crash-on-failure-or-not" thing?

Reply:
tja, upon implementing it felt quite strange to generally 'try/except' each and then for `Machine` to care about how to handle exceptions by checking `Reporter` for some 'thing'. 

Rather, IMO, handle it during the implementation of a `Reporter`. As such, if it should be allowed to fail, then _it_ should rather take some 'hardfail' flag as a parameter and set per config entry.